iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -423,6 +423,28 @@ static LogicalResult analyseDestructiveUpdateOp(Operation *op, Value source,
   return success();
 }
 
+static LogicalResult analyseScfForOp(scf::ForOp forOp,
+                                     BufferizationPlan &plan) {
+  if (forOp.results().empty()) return success();
+  auto yeildOp = cast<scf::YieldOp>(forOp.getBody()->getTerminator());
+  for (int i = 0; i < yeildOp.results().size(); ++i) {
+    Value outputTensor = yeildOp.results()[i];
+    Value resultTensor = forOp.results()[i];
+    Value initArg = forOp.initArgs()[i];
+    Value operand = forOp.getBodyRegion().getArgument(1);
+    plan.unionSets(outputTensor, resultTensor);
+    plan.unionSets(outputTensor, initArg);
+    plan.unionSets(outputTensor, operand);
+  }
+  return success();
+}
+
+static LogicalResult analyseTensorExtractOp(tensor::ExtractOp extractOp,
+                                            BufferizationPlan &plan) {
+  plan.unionSets(extractOp.tensor(), extractOp.result());
```

Comment:
I think we need a bit more care here. We want to only add `tensor` types to equivalence sets. So if the `result()` is not a tensor type, we can just convert a `tensor.extract` into a `memref.load` operation in the conversion.

Reply:
Ah goo catch! `results` is `AnyType` e.g. I think what we can do is to `union` if `result()` is a `TensorType` otherwise do nothing. Converting `tensor.extract` is done by  `convertTensorExtractOp`